paprika/align.py

```diff
@@ -192,3 +192,158 @@ def offset_structure(structure, offset):
     structure.coordinates = offset_coords
     logger.info("Added offset of {} to atomic coordinates...".format(offset))
     return structure
+
+
+def translate_to_origin(structure, weight="mass", atom_mask=None, dim_mask=None):
+    """Translate a structure to the origin based on the center of 
+       mass of the whole system or of a particular choice of atom(s).
+
+    Parameters
+    ----------
+    structure : str or parmed.Structure
+        Molecular structure containing coordinates.
+    weight : str
+        Calculate the center based on atomic masses ('mass' default) or geometric center ('geo')?
+    atom_mask : str
+        Selection of atom(s) if a particular subset is preferred to estimate the center.
+    dim_mask : list
+        A mask that will filter the dimensions to which the translation will be applied (default is [1,1,1]).
+
+    Returns
+    -------
+    structure : parmed.Structure
+        A molecular structure with the coordinates translated to the origin.
+
+    """
+    # Check if weight variable is properly chosen
+    if weight not in ["mass", "geo"]:
+        raise SystemExit('Error: "weight" must either be "mass" or "geo"')
+
+    # Dimension mask
+    if dim_mask is None:
+        dim_mask = [1, 1, 1]
+    elif len(dim_mask) != 3:
+        raise SystemExit(
+            'Error: "dim_mask" must be a list with 3 elements, e.g. [1,1,1]'
+        )
+    dim_mask = np.array(dim_mask)
+
+    # Atom coordinates and masses
+    if atom_mask is None:
+        coordinates = structure.coordinates
+        masses = np.asarray([atom.mass for atom in structure.atoms])
+    else:
+        coordinates = structure[atom_mask].coordinates
+        masses = np.asarray([atom.mass for atom in structure[atom_mask].atoms])
+
+    # Convert weight if geometric center
+    if weight == "geo":
+        masses[:] = 1.0
+
+    # Center of mass/geometry
+    centroid = pmd.geometry.center_of_mass(coordinates, masses)
+
+    # Translate coordinates
+    aligned_coords = np.empty_like(structure.coordinates)
+    for atom in range(len(structure.atoms)):
+        aligned_coords[atom] = structure.coordinates[atom] - centroid * dim_mask
+    structure.coordinates = aligned_coords
+
+    return structure
+
+
+def align_principal_axes(structure, atom_mask=None, princ_axis=None, v_axis=None):
+    """Aligns the a chosen principal axis of a system to a specified axis. 
+       This function is based on the method given in the link:
+
+        * https://www.ks.uiuc.edu/Research/vmd/script_library/scripts/orient/
+
+    Parameters
+    ----------
+    structure : parmed.Structure
+        Molecular structure containing coordinates.
+    atom_mask : str
+        A mask that filter specific atoms for calculating the moment of inertia.
+    princ_axis : int
+        Which principal axis to align? (choice: 1, 2 or 3, default: 1).
+    v_axis: list
+        The axis vector to align the system to (default: [0, 0, 1]).
```

Comment:
Does this mean `princ_axis` is going to be parallel to `v_axis`?

Reply:
yes